### PR TITLE
[graphql/rpc] available range support

### PR DESCRIPTION
**Pull request by oxade**
## Description 

According to data platform, the min and max checkpoints in checkpoint table correspond to the avail range, and the table is ordered.

## Test Plan 

e2e test

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/available_range.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::checkpoint::Checkpoint;
+use crate::context_data::db_data_provider::PgManager;
+use async_graphql::*;
+
+/// Information about whether epoch changes are using safe mode.
```
[the remark this answers is not in the document]
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/available_range.rs:8]: +1

File: crates/sui-graphql-rpc/src/types/query.rs [hunk 2]
```diff
@@ -37,6 +38,12 @@ impl Query {
             .extend()
     }
 
+    /// Range of checkpoints that the RPC has data available for (for data
+    /// that can be tied to a particular checkpoint).
+    async fn available_range(&self) -> Result<AvailableRange> {
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/query.rs:43]: nit: This doesn't need to return a `Result`.

**Review by amnn — COMMENTED** [review]
Some small nits, but looks great!

File: crates/sui-graphql-rpc/src/types/available_range.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (oxade) [reply crates/sui-graphql-rpc/src/types/available_range.rs:8]: Indeed its not
I created a followup fix but forgot to tag it here
https://github.com/MystenLabs/sui/pull/15228

**Pull request merged**